transitions/wave.h

```diff
@@ -56,5 +56,57 @@ class TrWaveX : public TransitionBaseX<FADEOUT_MS> {
     return MixColors(a, color_.getColor(led), mix, 15);
   }
 };
-   
+
+// TrSparkX generates a wave without Fade over the length of the blade from 
+// SPARK_CENTER. Unlike other transitions, this effect starts and ends
+// at the same color, and the wave is drawn using COLOR instead of the start/end
+// colors like most transitions to. It's intended to be used with TransitionLoopL
+// or TransitionEffectL, which takes transitions that start and begin with the same
+// color.
+
+template<class COLOR,
+         class SPARK_SIZE = Int<100>,
+         class SPARK_MS = Int<400>,
+         class SPARK_CENTER = Int<16384>>
+class TrSparkX : public TransitionBaseX<SPARK_MS> {
+public:
+
+  void run(BladeBase *blade) {
+    spark_size_.run(blade);
+    spark_center_.run(blade);
+    spark_ms_.run(blade);
+    color_.run(blade);
+    if (this->restart()) {
+      center_ = spark_center_.getInteger(0);
+      size_ = spark_size_.getInteger(0);
+    }
+    TransitionBaseX<SPARK_MS>::run(blade);
+    num_leds_ = blade->num_leds();
+    offset_ = (millis() - this->start_millis()) * 32768 / spark_ms_.getInteger(0);
```

Comment:
I think we can replace this with offset_ = update(32768);

Reply:
updated